### PR TITLE
seperate OjoTrack and argv handling

### PR DESCRIPTION
**Pull request by FreaxMATE**
This seperates OjoTrack to ojo-track.c (with ojo-track.h) and adds the ability to handle command line arguments. As follows you can now right click on multiple media files and open with ojo from your favorite file manager.

### CODE REVIEW

File: src/ojo-player.c [hunk 1]
```diff
@@ -85,29 +85,41 @@ void ojo_player_tracks_initialize (GSList *list, int n_tracks, gboolean add)
...
       {
-         ojo_player->tracks[track_index]->type = -1 ;
+         ojo_track_set_type(ojo_player->tracks[track_index], UNKNOWN) ;
          fprintf (stderr, "WARNING: open_media() in vlcPlayer.c: media type neither audio nor video\n") ;
          return ;
       }
       libvlc_media_tracks_release(libvlc_tracks[0], n_streams) ;
-      g_free(list->data) ;
       track_index++ ;
+      // g_free(list->data) ; TODO: free uris; causes errors
```
Comment (FreaxMATE) [inline src/ojo-player.c]: When uncommented, this causes some troubles (freeing the uri strings (gpointer) returned by file manager)

**Review by mbkma — CHANGES_REQUESTED** [review]
Title is "unknown" when opening from right-click menu for me. The uri strings should be freed before we merge this. Otherwise it works.

File: src/ojo-player.c (hunk 1 above)
Comment (FreaxMATE) [reply src/ojo-player.c]: solved in 56e5ba5

**Comment by FreaxMATE** [comment]
Everything should be fine now.

**Comment by mbkma** [comment]
In this branch I get an error when opening ~20 music files. gdb:
```
.
.
.
WARNING: ojo_player_tracks_initialize() in ojo-player.c: failed to fetch metadata
WARNING: ojo_player_tracks_initialize() in ojo-player.c: failed to fetch metadata
WARNING: ojo_player_tracks_initialize() in ojo-player.c: failed to fetch metadata
WARNING: ojo_player_tracks_initialize() in ojo-player.c: failed to fetch metadata
WARNING: ojo_player_tracks_i^C
Thread 1 "ojo" received signal SIGINT, Interrupt.
__GI___libc_write (nbytes=82, buf=0x55555555d138, fd=2)
    at ../sysdeps/unix/sysv/linux/write.c:26
26	../sysdeps/unix/sysv/linux/write.c: No such file or directory.
(gdb) backtrace
#0  0x00007ffff7251537 in __GI___libc_write
    (nbytes=82, buf=0x55555555d138, fd=2)
    at ../sysdeps/unix/sysv/linux/write.c:26
#1  0x00007ffff7251537 in __GI___libc_write
    (fd=2, buf=0x55555555d138, nbytes=82)
    at ../sysdeps/unix/sysv/linux/write.c:24
#2  0x00007ffff71e23bd in _IO_new_file_write
    (f=0x7ffff7323680 <_IO_2_1_stderr_>, data=0x55555555d138, n=82)
    at fileops.c:1183
#3  0x00007ffff71e175f in new_do_write
    (fp=fp@entry=0x7ffff7323680 <_IO_2_1_stderr_>, data=data@entry=0x55555555d138 "WARNING: ojo_player_tracks_initialize() in ojo-player.c: failed to fetch metadata\n", to_do=to_do@entry=82) at libioP.h:839
#4  0x00007ffff71e2aee in _IO_new_file_xsputn
    (n=82, data=<optimized out>, f=0x7ffff7323680 <_IO_2_1_stderr_>)
    at fileops.c:1262
#5  0x00007ffff71e2aee in _IO_new_file_xsputn
    (f=0x7ffff7323680 <_IO_2_1_stderr_>, data=<optimized out>, n=82)
    at fileops.c:1204
#6  0x00007ffff71d7ae8 in __GI__IO_fwrite
    (buf=0x55555555d138, size=1, count=82, fp=0x7ffff7323680 <_IO_2_1_stderr_>)
    at libioP.h:839
#7  0x0000555555559cd7 in ojo_player_tracks_initialize
```

**Review by mbkma — APPROVED** [review]
Its fine.

**Pull request merged**